content/post/deploy-elasticsearch-in-gke/index.md

```diff
@@ -0,0 +1,686 @@
+health status index         uuid                   pri rep docs.count docs.deleted store.size pri.store.size
+green  open   kubedb-system zeIvl0RAQ6uhhcvW5GoWfg   1   1          1            5        1mb        516.3kb
+```
+
+### Create a RestoreSession
+
+Below, is the contents of YAML file of the `RestoreSession` object that we are going to create.
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: es-topology-restore
+  namespace: demo
+spec:
+  repository:
+    name: gcs-repo
+  target:
+    ref:
+      apiVersion: appcatalog.appscode.com/v1alpha1
+      kind: AppBinding
+      name: es-topology-cluster 
+  rules:
+    - snapshots: [latest]
+```
+
+Now, let's create `RestoreSession` that will initiate restoring from the cloud.
+
+```bash
+$ kubectl apply -f es-topology-restore.yaml 
+restoresession.stash.appscode.com/es-topology-restore created
+```
+
+This `RestoreSession` specifies where the data will be restored.
+Once this is applied, a `RestoreSession` will be created. Once it has succeeded, the database has been successfully recovered as you can see below:
+
+```bash
+$ kubectl get restoresession -n demo
+NAME                  REPOSITORY   PHASE       DURATION   AGE
+es-topology-restore   gcs-repo     Succeeded   10s        27s
+```
+
+Now, let's check whether the database has been correctly restored:
+
+```bash
+$ curl -XGET -k --user 'elastic:q)UC;l!!euEnk.ZZ' "https://localhost:9200/_cat/indices?v&s=index&pretty"
+health status index         uuid                   pri rep docs.count docs.deleted store.size pri.store.size
+green  open   kubedb-system zeIvl0RAQ6uhhcvW5GoWfg   1   1          1            5        1mb        516.3kb
+green  open   music         nZ_6Se_gQoOPDYVpYmeVFw   1   1          1            0     10.5kb          5.2kb
+```
+Also, let’s verify the data in the indexes:
+
+```bash
+$ curl -XGET -k --user 'elastic:q)UC;l!!euEnk.ZZ' "https://localhost:9200/music/_search?pretty"
+{
+  "took" : 9,
+  "timed_out" : false,
+  "_shards" : {
+    "total" : 1,
+    "successful" : 1,
+    "skipped" : 0,
+    "failed" : 0
+  },
+  "hits" : {
+    "total" : {
+      "value" : 1,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "music",
+        "_id" : "oyO52YYBpvYWEhqZyPph",
+        "_score" : 1.0,
+        "_source" : {
+          "Artist" : "Bon Jovi",
+          "Song" : "Its My Life"
+        }
+      }
+    ]
+  }
+}
+
+
+```
+
+> You can see the database has been restored. The recovery of Elasticsearch has been successful. If you faced any difficulties in the recovery process, you can reach out to us through [EMAIL](mailto:support@appscode.com?subject=Stash%20Recovery%20Failed%20in%20GKE).
+
+We have made an in depth tutorial on OpenSearch OpsRequests - Day 2 Lifecycle Management for OpenSearch Cluster Using KubeDB. You can have a look into the video below:
```

Comment:
Opensearch video is irrelavant here. Use this video instead. Here, we have deployed Elasticsearch 7.14.0 in the Linode cluster.